docs/Caliptra_rtl.md

```diff
@@ -0,0 +1,862 @@
+
+For more information, see[ Caliptra: A Datacenter System on a Chip (SoC) Root of Trust (RoT)](https://chipsalliance.github.io/Caliptra/doc/Caliptra.html).
+
+## References and related specifications
+
+The blocks described in this document are either obtained from open-source GitHub repositories, developed from scratch, or modification of open-source implementations. Links to relevant documentation and GitHub sources are shared in the following table.
+
+*Table 1: Related specifications*
+
+| IP/Block | GitHub URL | Documentation | Link |
+| :--------- | :--------- | :--------- |:--------- |
+| Cores-VeeR | [GitHub - chipsalliance/Cores-VeeR-EL2](https://github.com/chipsalliance/Cores-VeeR-EL2) | VeeR EL2 Programmer’s Reference Manual | [chipsalliance/Cores-VeeR-EL2 · GitHubPDF](http://cores-swerv-el2/RISC-V_SweRV_EL2_PRM.pdf%20at%20master%20%C2%B7) |
+| AHB Lite Bus | [aignacio/ahb_lite_bus: AHB Bus lite v3.0 (github.com)](https://github.com/aignacio/ahb_lite_bus) | AHB Lite Protocol<br> [Figure 2: SoC interface block diagram](#soc-interface-definition) | [ahb_lite_bus/docs at master · aignacio/ahb_lite_bus (github.com)](https://github.com/aignacio/ahb_lite_bus/tree/master/docs)<br> [ahb_lite_bus/diagram_ahb_bus.png at master · aignacio/ahb_lite_bus (github.com)](https://github.com/aignacio/ahb_lite_bus/blob/master/diagram_ahb_bus.png) |
+| SHA 256 | [secworks/sha256: Hardware implementation of the SHA-256 cryptographic hash function (github.com)](https://github.com/secworks/sha256) | | |
+| SHA 512 | | | |
+| SPI Controller | <https://github.com/pulp-platform/axi_spi_master> | | |
+
+# Caliptra Core
+
+The following figure shows the Caliptra Core.
+
+*Figure 1: Caliptra block diagram*
+
+## Boot Media Dependent (passive) vs Boot Media Integrated (active) profile
+
+In the BMD profile, QSPI and I3C IO peripherals are disabled using integration-time defines passed to the hardware, which are also exposed to ROM. Peripheral IOs can be tied off appropriately for the BMD profile at SoC integration time. For more information on the passive vs. active profile differences, see the boot flows in[ Caliptra profiles](https://chipsalliance.github.io/Caliptra/doc/Caliptra.html#caliptra-profiles). Only the BMD profile is supported for the first generation release of Caliptra.
+
+# SoC interface definition
+
+The following figure shows the SoC interface definition.
+
+*Figure 2: SoC Interface Block Diagram*
+
+## Integration parameters
+
+The following table describes integration parameters.
+
+*Table 2: Integration parameters*
+
+| **Parameter name** | **Width** | **Description** |
+| :--------- | :--------- | :--------- |
+| APB_ADDR_WIDTH | 32 | Width of the APB Address field. Default to 32. |
+| APB_DATA_WIDTH | 32 | Width of the APB Data field. Default to 32. |
+| APB_USER_REQ_WIDTH | TODO | Width of the APB PAUSER field. |
```

Comment:
Can we remove TODOs? Or create an issue to track removal of them and reference the Issue here.

Reply:
issue created #260

Reply:
For this specific TODO - the value is '32'.
The issue is still useful for other todos like the LINT rules though.